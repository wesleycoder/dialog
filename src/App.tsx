import { useRef } from 'react'
import styles from './App.module.css'
import { Dialog, DialogHandle } from './components/dialog'

function App() {
  const dialogRef = useRef<DialogHandle>(null)

  return (
    <>
      <div className={styles.content}>
        <button
          className={styles.openDialog}
          onClick={() => {
            dialogRef.current?.showModal()
          }}
        >
          Open dialog
        </button>
        <Dialog
          ref={dialogRef}
          title="Lemon"
        >
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit?
          Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum
          porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam
          deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex,
          provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem
          consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo
          dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur
          adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora
          dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit
          amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum
          officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi!
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit?
          Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum
          porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam
          deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex,
          provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem
          consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo
          dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur
          adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora
          dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit
          amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum
          officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi!
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit?
          Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum
          porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam
          deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex,
          provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem
          consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo
          dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur
          adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora
          dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit
          amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum
          officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi!
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit?
          Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum
          porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam
          deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex,
          provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem
          consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo
          dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur
          adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora
          dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit
          amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum
          officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi!
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit?
          Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum
          porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam
          deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex,
          provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem
          consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo
          dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur
          adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora
          dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit
          amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum
          officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi!
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit?
          Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum
          porro fugiat commodi!
        </Dialog>
      </div>
    </>
  )
}

export default App
