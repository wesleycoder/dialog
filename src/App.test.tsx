import { describe, expect, it } from 'vitest'
import { render } from '../config/test/render'
import App from './App'

describe('App', () => {
  it('should render', () => {
    const { baseElement } = render(<App />)
    expect(baseElement).toMatchSnapshot()
  })
})
