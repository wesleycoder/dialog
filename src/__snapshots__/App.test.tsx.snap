// Vitest Snapshot v1

exports[`App > should render 1`] = `
<body>
  <div>
    <div
      class="_content_6dc87e"
    >
      <button
        class="_openDialog_6dc87e"
      >
        Open dialog
      </button>
      <dialog
        class="_dialog_40d523"
      >
        <header
          class="_header_40d523"
        >
          <h2
            class="_title_40d523"
          >
            Lemon
          </h2>
          <button
            class="_close_40d523"
          >
            <svg
              fill="none"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 1L8 8L15 15"
                stroke="black"
                stroke-width="1.2"
              />
              <path
                d="M1 1L8 8L0.999999 15"
                stroke="black"
                stroke-width="1.2"
              />
            </svg>
          </button>
        </header>
        <div
          class="_content_40d523"
        >
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi! Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem consequatur libero totam deleniti suscipit? Repellendus voluptatum officia, in, tempora dignissimos illo dolorem necessitatibus eos ex, provident rerum porro fugiat commodi!
        </div>
      </dialog>
    </div>
  </div>
</body>
`;
